src/centroiders.cpp

```diff
@@ -120,6 +118,127 @@ std::vector<Star> CenterOfGravityAlgorithm::Go(unsigned char *image, int imageWi
+            p.xMax = i % imageWidth;
+        } else if (i % imageWidth < p.xMin) {
+            p.xMin = i % imageWidth;
+        }
+        if (i / imageWidth > p.yMax) {
+            p.yMax = i / imageWidth;
+        } else if (i / imageWidth < p.yMin) {
+            p.yMin = i / imageWidth;
+        }
+        if(i % imageWidth != imageWidth - 1) {
+            IWCoGHelper(p, i + 1, image, imageWidth, imageHeight, starIndices);
+        }
+        if (i % imageWidth != 0) {
+            IWCoGHelper(p, i - 1, image, imageWidth, imageHeight, starIndices);
+        }
+        IWCoGHelper(p, i + imageWidth, image, imageWidth, imageHeight, starIndices);
+        IWCoGHelper(p, i - imageWidth, image, imageWidth, imageHeight, starIndices);
+    }
+}
+
+std::vector<Star> IterativeWeightedCenterOfGravityAlgorithm::Go(unsigned char *image, int imageWidth, int imageHeight) const {
+    IWCoGParams p;
+    std::vector<Star> result;
+    p.cutoff = DetermineCutoff(image, imageWidth, imageHeight);
+    for (int i = 0; i < imageHeight * imageWidth; i++) {
+        //check if pixel is part of a "star" and has not been iterated over
+        if (image[i] >= p.cutoff && p.checkedIndices.count(i) == 0) {
+            std::vector<int> starIndices; //indices of the current star
+            p.maxIntensity = 0;
+            int xDiameter = 0; 
+            int yDiameter = 0;
+            float yWeightedCoordMagSum = 0; 
+            float xWeightedCoordMagSum = 0; 
+            float weightedMagSum = 0; 
+            float fwhm; //fwhm variable
+            float standardDeviation;
+            float w; //weight value
+            
+            //computes indices to skip after done w current star
+            int step = i;
+            while(step < imageWidth * imageHeight && (step + 1) % imageWidth != 0 && image[step] >= p.cutoff) {
+                step++;
+            }
+
+            p.xMax = i % imageWidth;
+            p.xMin = i % imageWidth;
+            p.yMax = i / imageWidth;
+            p.yMin = i / imageWidth;
+
+            IWCoGHelper(p, i, image, imageWidth, imageHeight, starIndices);
+
+            xDiameter = (p.xMax - p.xMin) + 1;
+            yDiameter = (p.yMax - p.yMin) + 1;
+
+            //calculate fwhm
+            int count = 0;
+            for (int j = 0; j < (int) starIndices.size(); j++) {
+                if (image[starIndices.at(j)] > p.maxIntensity / 2) {
+                    count++;
+                }
+            }
+            fwhm = sqrt((float) count);
+            standardDeviation = fwhm / (2.0 * sqrt(2.0 * log(2.0)));
+            float guessXCoord = (float) (p.guess % imageWidth);
+            float guessYCoord = (float) (p.guess / imageWidth);
+            //how much our new centroid estimate changes w each iteration
+            float change = 100.0;
+            //while we see some large enough change in estimated, maybe make it a global variable
+            while (change > 0.0001) {
```

Comment:
Maybe add it as a private member of the `IterativeWeightedCenterOfGravityAlgorithm` class, then prompt for it in `IWCoGCentroidAlgorithmPrompt`?